### ISSUE TITLE
[Doc] max_tree_depth LGBM

### ISSUE THREAD

**Issue by partiallycomplex** [ThomasBury/arfs#10]
Hi I wonder if it is necessary to limit the tree depth, as in boruta, and how much this parameter (and in general all LGBM settings, leaves, boosting rounds...) could impact BoostAGroota. Is it worth including in the doc ?
Thanks, great project

**Comment by ThomasBury** [comment]
Hi @partiallycomplex 

Thank you for your kind words. 

It is indeed a good idea the limit the depth, to avoid using "specific" models in the selection process. The idea is to avoid overfitting and keep the default parameters of lightGBM as is. The authors said that the default values guarantee sane but good behaviour in most cases.

For instance, the depth is not restricted (-1) but the max number of leaves is set to 31. In most cases, it avoids overfitting. The number of estimators is by default set to 100. You could decrease it a bit if you think it leads to overfitting. 

The only parameters I'd change are: 
- the `loss` (according to the task)
- the `sample_weight` (if any)
- the `class_weight` if dealing with unbalance classification.

 You can also compare the results to `GrootCV` which is more robust (cross-validated). 
 I'll try to add this in the doc, which is now live on ReadTheDocs btw ^^

**Comment by partiallycomplex** [comment]
Hi @ThomasBury,
I agree with leaving default parameters, except what you mentioned. Actually class_weight could be skipped by setting is_unbalance=True in the LGBMClassifier

**Comment by ThomasBury** [comment]
> Hi @ThomasBury,
> I agree with leaving default parameters, except what you mentioned. Actually class_weight could be skipped by setting is_unbalance=True in the LGBMClassifier 
> 

Yes, both can be used for unbalanced classification. Whatever works best for your project 🙂

**Comment by notuntoward** [comment]
Hi @ThomasBury, class_weight is needed for multi-class classification, though.

**Comment by ThomasBury** [comment]
> Hi @ThomasBury, class_weight is needed for multi-class classification, though.

it is, you can use it without any problem for multi-class (Even for binary + imbalance, I'd personally use `class_weight`, it often leads to better results with a custom weighting vs the class proportional weighting of `is_unbalanced`)
